Rust compiler requires python 2.7

The "Building from Source" instructions on the github wiki page specify that the rust compiler requires
python 2.6 or later, however the ./configure script says that python 2.7 is required.

Thanks for following up about this.

     I found this warning about updating python.  I'm running CentOS 6.2
presently:
CentOS 6.2 ships with Python 2.6.6 and depends on that specific version. Be
careful not to replace it or bad things will happen. If you need access to
a newer version of Python you must compile it yourself and install it
side-by-side with the system version.
source:
https://github.com/h2oai/h2o-2/wiki/Installing-python-2.7-on-centos-6.3.-Follow-this-sequence-exactly-for-centos-machine-only

    Here are the last few lines of output I get from the rust configure
script:
checking for python... user defined: /usr/bin/python2.6
checking for python >= 2.7... not found
configure: error: found python 2.6.6 (/usr/bin/python2.6); required >= 2.7
See `config.log' for more details.
configure: error: LLVM configure failed


On Wed, Dec 30, 2015 at 5:32 AM, rkruppe <notifications@github.com> wrote:

> Our Python situation is really confusing. The configure script indicates
> that Python is needed for LLVM, but LLVM's docs say it only needs Python
> for its tests, and we (thankfully) don't run LLVM tests. Perhaps there is
> some flag that enables them, but by default that is not the case.
> Furthermore, configure seems to accept Python versions as old as 2.4 (or
> rather, rejects 2.3 because "LLVM requires Python 2.4-2.7") when provided
> with a --python=<path> argument, though as @tshepang
> <https://github.com/tshepang> found it will look for only 2.6/2.7 in the
> absence of this argument. But LLVM 2.8's docs state that Python ">=2.7" is
> required (from lurking on the llvm-dev mailing list, I recall that this
> does *not* include 3.x).
>
> As if all that wasn't bad enough, we have a number of Python scripts in
> src/etc/ with no easy way to tell what Python version those require. At a
> glance, I saw one use of the with statement which gives us a lower bound
> of 2.6, and it's quite possibly that some 2.7 features have slipped in over
> the years. Two scripts that I wrote explicitly require 2.7 but that's
> unavoidable and those scripts are not part of the normal build process of
> even most development work. The others, no idea.
>
> Honestly I would be in favor of just requiring 2.7 regardless of whether
> anything even needs it. Just to make all this confusion stop.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rust-lang/rust/issues/30618#issuecomment-167974469>.
>

--- a/README.md
+++ b/README.md
@@ -17,7 +17,7 @@ Read ["Installing Rust"] from [The Book].
 1. Make sure you have installed the dependencies:
 
    * `g++` 4.7 or `clang++` 3.x
-   * `python` 2.6 or later (but not 3.x)
+   * `python` 2.7 or later (but not 3.x)
    * GNU `make` 3.81 or later
    * `curl`
    * `git`

